src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else if (size == 1) {
+            Node newNode = new Node(head, value, null);
+            head.next = newNode;
+            tail = newNode;
+        } else {
+            Node newNode = new Node(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheckForAdd(index);
+        Node<T> indexNode = findNodeByIndex(index);
+        if (indexNode == null) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            Node newNode = new Node(null, value, head);
+            head.prev = newNode;
+            head = newNode;
+            size++;
+            return;
+        }
+        Node newNode = new Node(indexNode.prev, value, indexNode);
+        indexNode.prev.next = newNode;
+        indexNode.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexCheck(index);
+        Node<T> indexNode = findNodeByIndex(index);
+        T oldValue = indexNode.item;
+        indexNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexCheck(index);
+        Node<T> removedElement = findNodeByIndex(index);
+        unlink(removedElement);
+        return removedElement.item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int index = findIndexByValue(object);
+        if (index == -1) {
+            return false;
+        } else {
+            unlink(findNodeByIndex(index));
+        }
```

Comment:
```suggestion
         unlink(findNodeByIndex(index));
```
we don't need else after returning/throwing